web/app/Models/UserMessage.php

```diff
@@ -0,0 +1,356 @@
+    }
+
+    /**
+     * Retrieve draft messages.
+     * @param Builder $query
+     * @param User $user
+     * @return Builder
+     */
+    public function scopeDrafts(Builder $query, User $user) : Builder
+    {
+        return $query->where('from_user_id', $user->id)
+            ->where('flags', '&', self::MESSAGE_DRAFT);
+    }
+
+    /**
+     * Retrieve archived messages.
+     * @param Builder $query
+     * @param User $user
+     * @return Builder
+     */
+    public function scopeArchive(Builder $query, User $user) : Builder
+    {
+        return $query->where('to_user_id', $user->id)
+        ->where('flags', '&', self::MESSAGE_ARCHIVED);
+    }
+
+    /**
+     * Retrieve starred messages.
+     * @param Builder $query
+     * @param User $user
+     * @return Builder
+     */
+    public function scopeStarred(Builder $query, User $user) : Builder
+    {
+        return $query->where('to_user_id', $user->id)
+            ->where('flags', '&', self::MESSAGE_STARRED);
+    }
+
+    /**
+     * Retrieve sent messages. The only distinction is a MESSAGE_SENT flag.
+     * @param Builder $query
+     * @param User $user
+     * @return Builder
+     */
+    public function scopeSent(Builder $query, User $user) : Builder
+    {
+        return $query->where('from_user_id', $user->id)
+            ->where('flags', '&', self::MESSAGE_SENT);
+    }
+
+    /**
+     * Checks for the presence or absence of a particular flag.
+     */
+
+    /**
+     * Is this message read?
+     * @return bool
+     */
+    public function isRead() : bool
+    {
+        return $this->getFlag(UserMessage::MESSAGE_READ);
+    }
+
+    /**
+     * Is this message unread?
+     * @return bool
+     */
+    public function isUnread() : bool
+    {
+        return ! $this->getFlag(UserMessage::MESSAGE_READ);
+    }
+
+    /**
+     * Is this message a draft?
+     * @return bool
+     */
+    public function isDraft() : bool
+    {
+        return $this->getFlag(UserMessage::MESSAGE_DRAFT);
+    }
+
+    /**
+     * Is this message not a draft?
+     * @return bool
+     */
+    public function isNotDraft() : bool
+    {
+        return ! $this->getFlag(UserMessage::MESSAGE_DRAFT);
+    }
+
+    /**
+     * Is this message starred?
+     * @return bool
+     */
+    public function isStarred() : bool
+    {
+        return $this->getFlag(UserMessage::MESSAGE_STARRED);
+    }
+
+    /**
+     * Is this message unstarred?
+     * @return bool
+     */
+    public function isUnstarred() : bool
+    {
+        return ! $this->getFlag(UserMessage::MESSAGE_STARRED);
+    }
+
+    /**
+     * Is this message archived?
+     * @return bool
+     */
+    public function isArchived() : bool
+    {
+        return $this->getFlag(UserMessage::MESSAGE_ARCHIVED);
+    }
+
+    /**
+     * Is this message not archived?
+     * @return bool
+     */
+    public function isNotArchived() : bool
+    {
+        return ! $this->getFlag(UserMessage::MESSAGE_STARRED);
```

Comment:
Probably not, I wanted the ones for `isRead` and `isUnread` because they'll have roughly equal utility and I think adding the negation in that way isn't as clean. I may have gone overboard for things like Archived and Sent.